src/AoC2024/1/aoc24_1.cpp

```diff
@@ -1,7 +1,66 @@
-#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <string>
 
-int main(){
-    std::cout << "Hello world!" << '\n';
+#include "aoc24_1.h"
 
-    return 0;
+std::vector<int>
+calculateDistances(const std::vector<std::pair<int,int>>& preparedList) {
+    std::vector<int> distances = {};
+
+    const bool listIsNotEmpty = preparedList.size() > 0;
+    if(listIsNotEmpty)
+        std::for_each(preparedList.begin(), preparedList.end(), [&](const std::pair<int,int>& pair){
+            int left = std::get<0>(pair);
+            int right = std::get<1>(pair);
+            int distance = std::abs(left - right);
+            distances.push_back(distance);
+        });
+    
+    return distances;
+}
+
+std::pair<std::vector<int>, std::vector<int>>
+readPairsFromFile(const std::filesystem::path& inputfile){
+    std::vector<int> left = {};
+    std::vector<int> right = {};
+    constexpr int expectedNumbeOfLines = 1000;
+    left.reserve(expectedNumbeOfLines);
+    right.reserve(expectedNumbeOfLines);
+
+    std::string line = {""};
+    int num1 = 0;
+    int num2 = 0;
```

Comment:
hard RAII would move these instantiations down right before the while.  or even into the while: `while ( int a = 0; a < 100 ) { …` , but this style is highly preference dependent.